src/azure/encrypt_payload/encrypt_payload.cc

```diff
@@ -0,0 +1,310 @@
+      key_fetcher_manager->GetPrivateKey(std::to_string(key_id));
+  CHECK(private_key.has_value()) << "Failed to fetch private key";
+
+  std::string ciphertext_bytes;
+  CHECK(privacy_sandbox::azure_encrypt_payload::Base64Decode(
+            encrypt_payload_result.ciphertext_base64, ciphertext_bytes)
+            .Successful())
+      << "Failed to encode ciphertext as base64";
+
+  server_common::EncapsulatedRequest encapsulatedRequest = {ciphertext_bytes,
+                                                            label_for_test};
+
+  absl::StatusOr<quiche::ObliviousHttpRequest> decrypted_result =
+      server_common::DecryptEncapsulatedRequest(private_key.value(),
+                                                encapsulatedRequest);
+  CHECK(decrypted_result.ok()) << "Decryption failed";
+  return std::string(decrypted_result->GetPlaintextData());
+}
+}  // namespace
+}  // namespace privacy_sandbox::azure_encrypt_payload
+
+int main(int argc, char** argv) {
+  absl::ParseCommandLine(argc, argv);
+
+  // Setup
+  const auto get_token_url = absl::GetFlag(FLAGS_get_token_url);
+  setenv("AZURE_BA_PARAM_GET_TOKEN_URL", get_token_url.c_str(), 0);
+
+  const auto unwrap_private_key_endpoint =
+      absl::GetFlag(FLAGS_unwrap_private_key_endpoint);
+  setenv("AZURE_BA_PARAM_KMS_UNWRAP_URL", unwrap_private_key_endpoint.c_str(),
+         0);
+
+  // grpc_init initializes and shutdown gRPC client.
+  privacy_sandbox::server_common::GrpcInit grpc_init;
+  google::scp::cpio::CpioOptions cpio_options;
+  cpio_options.log_option = google::scp::cpio::LogOption::kConsoleLog;
+  CHECK(google::scp::cpio::Cpio::InitCpio(cpio_options).Successful())
+      << "Failed to initialize CPIO library";
+
+  auto key_fetcher_manager =
+      privacy_sandbox::azure_encrypt_payload::CreateKeyFetcherManager(
+          privacy_sandbox::azure_encrypt_payload::CreatePublicKeyFetcher());
+
+  bool decrypt_mode = {absl::GetFlag(FLAGS_decrypt_mode)};
+
+  if (!decrypt_mode) {
+    std::string plaintext_payload;
+    std::cin >> plaintext_payload;
+    const auto encrypt_result =
+        privacy_sandbox::azure_encrypt_payload::EncryptPayload(
+            std::move(key_fetcher_manager), plaintext_payload);
+
+    const auto output_path = absl::GetFlag(FLAGS_output_path);
+    // We can improve this by checking the directly of the file exists.
+    // Currently it silently fails to write if the dir is not there.
+    std::ofstream fout(output_path);
+    fout << nlohmann::json(encrypt_result).dump(2);
+
+  } else {
+    nlohmann::json input;
+    std::cin >> input;
+    auto encryption_result = input.template get<
+        privacy_sandbox::azure_encrypt_payload::EncryptPayloadResult>();
+    const auto output_path = absl::GetFlag(FLAGS_output_path);
+    // We can improve this by checking the directly of the file exists.
+    // Currently it silently fails to write if the dir is not there.
+    std::ofstream fout(output_path);
```

Comment:
I should be able to fix it with https://en.cppreference.com/w/cpp/filesystem/create_directory